### PR TITLE
Improve image transcription

### LINKED ISSUE #27: Bot is not able to transcribe text within image in some cases
**Describe the bug**
In some cases the bot is not able to discern what the text is within an image. Either it can only see some of the text within an image or none at all.

**To Reproduce**
1. Run `textifyBot.py`
2. See that some text is transcribed in broken English or unintelligible

**Expected behavior**
The bot is expected to transcribe the text within an image accurately.

**Additional context**
There are some potential fixes out there. Post processing the image to make the text easier to see against the background is an option. Bluring the background or making the image black and white have produced good results in other projects.

### PR DESCRIPTION
**Pull request by baconbm**
This PR utilizes OpenCV to improve the image transcription process. It enhances images by enlarging them slightly, making it easier for the OCR library to see the text within the image. Closes #27

### CODE REVIEW

**Comment by afig** [comment]
I think sticking with just the image resizing for now is a good idea.

Is there any particular reason you chose to use OpenCV to do the image resizing? Pillow, the fork of PIL (Python Image Library) that we are already using seems to have an [image resize feature](https://pillow.readthedocs.io/en/3.1.x/reference/Image.html#PIL.Image.Image.resize) as well.

Unless we need OpenCV's implementation for something, we may want to use Pillow/PIL's implementation to avoid a new dependency. (OpenCV is also a huge library whose main purpose is to do something completely different, and so using it for just image resizing seems misleading and inefficient).

Also, it seems like the unit tests for ExtractURL are being deleting in this PR (`testExtractURL.py`). Is this intentional?

**Comment by baconbm** [comment]
I agree @afig, while cv2 does have some cool features within it, importing such a big library for a small implementation is overkill. Perhaps we can find some use for it in the future, but for the current processing it is unnecessary. I replaced it with the `resize` method provided by `PIL`. 

The `testExtractURL.py` was the testing file I was using before we implemented unit testing, and the function that was being tested within it was made redundant by the unit tests that were implemented later, so I removed it within this PR.

File: src/TextifyBot.py [hunk 1]
```diff
@@ -9,7 +9,9 @@
 import praw
 import PIL
 import pytesseract
-import time
+from PIL import Image
+Import time
```
Comment (afig) [inline src/TextifyBot.py]:
```suggestion
import time
```

File: src/TextifyBot.py [hunk 2]
```diff
@@ -74,12 +76,23 @@ def allowedToParse(postID):
...
 
 def tesseractTranscribe(imagePath):
-    image = PIL.Image.open(imagePath)
+    image = PIL.Image.open(improveImage(imagePath))
     pytesseract.pytesseract.tesseract_cmd = TESSERACT_PATH
 
     return pytesseract.image_to_string(image)
 
 
+# converts downloaded image to grayscale
```
Comment (afig) [inline src/TextifyBot.py]:
```suggestion
```

File: src/TextifyBot.py [hunk 3]
```diff
@@ -74,12 +76,23 @@ def allowedToParse(postID):
...
+# and converts all pixels close to black to full black / all pixels close to white to full white
```
Comment (afig) [inline src/TextifyBot.py]:
```suggestion
# Makes the given image larger to improve success of transcription
```

**Review by afig — COMMENTED** [review]
Looks good. Just provided some suggestions to improve consistency of comments and imports.

**Review by afig — APPROVED** [review]
Looks good to merge, thanks @baconbm

**Review by calebeda — APPROVED** [review]

**Pull request merged**